### PR TITLE
Add number of touched files to commits

### LINKED ISSUE #116: Add number of touched files as a feature

### PR DESCRIPTION
**Pull request by ayush1999**
@marco-c Small question: Once I've made a change in repository.py, how do I update commits.json based on these changes?

Fixes #116.

### CODE REVIEW

**Comment by marco-c** [comment]
You will have to clone https://hg.mozilla.org/mozilla-central/ locally, then you can run `repository.py` giving it the path to your local clone.

File: bugbug/repository.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def _transform(commit):
         'bug_id': bug_id,
         'added': 0,
         'deleted': 0,
+        'touched':0,
```
Comment (marco-c) [inline bugbug/repository.py]: Call this one `files_modified_num`

File: bugbug/repository.py [hunk 2]
```diff
@@ -72,6 +73,7 @@ def _transform(commit):
         else:
             type_ = ext
         obj['types'].add(type_)
+        obj['touched'] = len(stats['touched'])
```
Comment (marco-c) [inline bugbug/repository.py]: `touched` is not the number of files modified, it is the number of lines modified.

**Comment by marco-c** [comment]
Can you try passing the absolute path to the repository?

**Comment by ayush1999** [comment]
I still get the same error.

**Comment by marco-c** [comment]
Can you `cd` into your cloned repository directory and type `hg log` and see if it works?

**Comment by ayush1999** [comment]
That's probably the reason. It works now, though I get a different error: `TypeError: __init__() got an unexpected keyword argument 'initializer'` is this because of some version issues?

**Comment by marco-c** [comment]
`initializer` was introduced in Python 3.7, so you'll need to use at least that version.

**Comment by marco-c** [comment]
Yep!

**Comment by marco-c** [comment]
(It takes a few hours, beware!)

**Comment by marco-c** [comment]
You'll also need to add a function in bug_extractor.py to get this feature.

**Comment by ayush1999** [comment]
> You'll also need to add a function in bug_extractor.py to get this feature.

You mean bug_features.py?

**Comment by marco-c** [comment]
> You mean bug_features.py?

Yes :)

**Comment by ayush1999** [comment]
@marco-c I just noticed that there's keras in the requirements, and this project uses python3.7, although there is no (official) support for tensorflow on python3.7 yet. Would this be a problem when a deep learning model might be tried out in the future?

**Comment by marco-c** [comment]
We can use tensorflow nightly which supports Python 3.7.

**Pull request merged**